azurerm/internal/services/storage/resource_arm_storage_account_customer_managed_key.go

```diff
@@ -60,9 +60,8 @@ func resourceArmStorageAccountCustomerManagedKey() *schema.Resource {
 			},
 
 			"key_version": {
-				Type:         schema.TypeString,
-				Required:     true,
-				ValidateFunc: validation.StringIsNotEmpty,
```

Comment:
According to the api documentation, the way it works is that so long as key_version is NOT in the properties pushed to the api, the key will auto rotate.
I understand the api does not accept an empty string for key_version.

Reply:
Hi @IanMoroney 

That's why we should keep this validation here.  Since this parameter is now optional, we should prevent it being an empty string if it is supplied so we don't send it to the API. Does that make sense?

Reply:
I have brought back the input validation and confirmed it works with the planned changes.
If key_version is specified, it doesn't accept an empty string.
If key_version is missing, it accepts it fine and enables automatic key rotation 👍